mbedtls/src/pk/mod.rs

```diff
@@ -138,6 +138,34 @@ const CUSTOM_PK_INFO: pk_info_t = {
     }
 };
 
+/// RSA components for constructing a private key.
+pub enum RsaPrivateComponents<'a> {
+    WithPrimes {
+        /// Private 1st prime
+        p: &'a Mpi,
+        /// Private 2nd prime
+        q: &'a Mpi,
+        /// Public exponent
+        e: &'a Mpi,
+    },
+    WithPrivateExponent {
+        /// Public modulus
+        n: &'a Mpi,
+        /// Private exponent
+        d: &'a Mpi,
+        /// Public exponent
+        e: &'a Mpi,
```

Comment:
I understand that these values are used by upstream in `rsa_import`, but `e` is redundant here.
What is the behavior if we pass inconsistent `d, e` here (with respect to `n`)? Perhaps we should add a test, since I don't see an explicit `d, n` check in `rsa_check_context` or in `rsa_complete` which we are rightfully calling on import. It looks like it assumes that `d, n` are consistent.

Reply:
I checked, and `rsa_check_context` does not guarantee consistency of parameters, and `rsa_complete` does not check this specifically.

Moreover I think it is an important check, for some implementations use `ed = 1 mod λ(n)` and others `ed = 1 mod φ(n)` (see e.g. https://gitlab.com/sequoia-pgp/sequoia/-/issues/792).

Reply:
After further research, MbedTLS returns `RSA_BAD_INPUT_DATA` when trying to `deduce_primes`, see [here](https://github.com/fortanix/rust-mbedtls/blob/75cb9b781fa240e4ef3ba3d49cd6da5efe7f225a/mbedtls-sys/vendor/library/rsa_internal.c#L160).

This check happens a bit too late and in principle inside a loop that aborts early or late depending on the input. Arguably this is upstream but we might want to check for consistency in this struct at least. I am now wondering if this loop be used by an attacker controlling `e, n` somehow to learn information about `d`, but this is upstream concern.

Reply:
addressed in https://github.com/fortanix/rust-mbedtls/pull/352